### ISSUE TITLE
Cannot build for iOS with Expo 47 using EAS due to outdated expo-linear-gradient dependency

### ISSUE THREAD

**Issue by bartvanandel** [adityamr15/react-native-wheel-picker-expo#11]
When building for iOS with this package as a dependency, EAS build fails with this error:

```
❌  (node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:4:8)

  2 | 
  3 | import CoreGraphics
> 4 | import ExpoModulesCore
    |        ^ compiling for iOS 12.0, but module 'ExpoModulesCore' has a minimum deployment target of iOS 13.0: /Users/expo/workingdir/build/ios/build/Build/Products/Debug-iphonesimulator/ExpoModulesCore/ExpoModulesCore.swiftmodule/x86_64-apple-ios-simulator.swiftmodule
  5 | 
  6 | public class LinearGradientModule: Module {
  7 |   public func definition() -> ModuleDefinition {
```

This is due to the fact that react-native-wheel-picker-expo relies on `expo-linear-gradient@^11.0.3`, which indeed targets iOS 12.4. This is fixed in version 12 of that library, specifically, [this commit](https://github.com/expo/expo/commit/c88b85cf3b619061827625853c5fc6b66e4c30c9), also check the [Changelog](https://github.com/expo/expo/commit/1327fe2bc92f71f7213cd5ddd40fa4f8a5336bc6#diff-e348afe46a16223b2c1704f39770f0f9e1254092f9ec6a94bd29cd212e4ed03a) from the release that followed it. Please consider upgrading your peer dependencies.

~~Alternatively, and probably better: make it a peer dependency instead of a direct dependency. Then you can specify a version range which includes both 11 and up, which will likely require less maintenance on your side if things change upstream.~~
whoops, just noticed it's already a peer dependency. Still, the version number could need an update.

**Comment by bartvanandel** [comment]
`expo-haptics` could also use an update, same problem with that package.

**Comment by haoyesoonEG** [comment]
+1

**Comment by adityamr15** [comment]
> `expo-haptics` could also use an update, same problem with that package.




> When building for iOS with this package as a dependency, EAS build fails with this error:
> 
> ```
> ❌  (node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:4:8)
> 
>   2 | 
>   3 | import CoreGraphics
> > 4 | import ExpoModulesCore
>     |        ^ compiling for iOS 12.0, but module 'ExpoModulesCore' has a minimum deployment target of iOS 13.0: /Users/expo/workingdir/build/ios/build/Build/Products/Debug-iphonesimulator/ExpoModulesCore/ExpoModulesCore.swiftmodule/x86_64-apple-ios-simulator.swiftmodule
>   5 | 
>   6 | public class LinearGradientModule: Module {
>   7 |   public func definition() -> ModuleDefinition {
> ```
> 
> This is due to the fact that react-native-wheel-picker-expo relies on `expo-linear-gradient@^11.0.3`, which indeed targets iOS 12.4. This is fixed in version 12 of that library, specifically, [this commit](https://github.com/expo/expo/commit/c88b85cf3b619061827625853c5fc6b66e4c30c9), also check the [Changelog](https://github.com/expo/expo/commit/1327fe2bc92f71f7213cd5ddd40fa4f8a5336bc6#diff-e348afe46a16223b2c1704f39770f0f9e1254092f9ec6a94bd29cd212e4ed03a) from the release that followed it. Please consider upgrading your peer dependencies.
> 
> ~Alternatively, and probably better: make it a peer dependency instead of a direct dependency. Then you can specify a version range which includes both 11 and up, which will likely require less maintenance on your side if things change upstream.~ whoops, just noticed it's already a peer dependency. Still, the version number could need an update.

Can't reproduce this issue from my end (no iOS device or simulator).
Appreciate if you open PR @bartvanandel

**Comment by bartvanandel** [comment]
I'm not behind the developer machine I encountered this with, nor do I currently have time to investigate further.  But since you're already having `"react": "*"` and `"react-native": "*"` in your peer deps in package.json, I'd suggest to simply allow the same wildcard for both other peer dependencies for the moment. Or, if you want to limit it to known compatible versions (which is 11 I guess for some people, as well as 12, which I've tested), use `"11-12"` for the both.
